backend/src/main/kotlin/com/example/demo/service/CardService.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.demo.service
+
+import com.example.demo.controller.dto.CreateCardRequest
+import com.example.demo.controller.dto.CardDTO
+import com.example.demo.model.Card
+import com.example.demo.model.User
+import com.example.demo.model.Ownership
+import com.example.demo.repo.CardRepository
+import com.example.demo.repo.OwnershipRepository
+import com.example.demo.repo.UserRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.domain.Pageable
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import java.util.*
+import java.util.stream.Collectors
+
+@Service
+class CardService(@Autowired private val cardRepository: CardRepository,
+                  @Autowired private val ownershipRepository: OwnershipRepository) {
+
+    public fun create(card: Card): Card {
+        return cardRepository.save(card)
+    }
+
+    public fun getById(id: Long): Optional<Card> {
+        return cardRepository.findById(id)
+    }
+
+    public fun getAll(pageable: Pageable): List<Card> {
+        val cardEntities = cardRepository.findAll(pageable)
+        val cards = cardEntities.map { it }
+        return cards.content
+    }
+    public fun changeName(id:Long, name:String): Card?{
+        val exists: Optional<Card> = getById(id)
+        if(exists.isPresent){
```

Comment:
usually endpoints to update an entity by id throw 404 if the entity is not present